sources/api/shibaken/src/main.rs

```diff
@@ -45,6 +49,13 @@ enum Commands {
     /// Fetch and return whether or not this host is in the given partition.
     /// Accepts multiple partitions, returning `true` if the host is in any of the given partitions.
     IsPartition(partition::IsPartition),
+
+    /// Fetch to IMDS to check if the instance on autoscaling warm pool is InService.
+    IsAutoscalingWarmPoolHealthy(warmpool::autoscaling_warm_pool::IsAutoscalingWarmPoolHealthy),
+
+    /// Support Bottlerocket refraining from joining a Kubernetes cluster if it's warming up in an autoscaling
+    /// warm pool until it is signaled to join the cluster.
+    BootShibaken(boot_shibaken::BootShibaken),
```

Comment:
Why two commands for this? I would just do it all in a single command. If there's a need to have a one-off check for status, you could take an argument, e.g.:
* `shibaken in-service-status`
* `shibaken in-service-status --wait`

But if you only need the "wait" behavior, `shibaken warm-pool-wait` is more succinct.